core/sup_tree_core/executor_pool/async_job_runner.ex

```diff
@@ -201,13 +203,16 @@ defmodule AntikytheraCore.ExecutorPool.AsyncJobRunner do
       end)
   end
 
+  defp decode_payload(payload) when is_binary(payload), do: :erlang.binary_to_term(payload)
+  defp decode_payload(payload)                        , do: payload
+
   #
   # Public functions to run within a separate process
   #
-  defun do_run(module :: v[module], payload :: v[map], metadata :: v[Metadata.t], context :: v[Context.t]) :: any do
+  defun do_run(module :: v[module], payload :: v[map | binary], metadata :: v[Metadata.t], context :: v[Context.t]) :: any do
     ContextHelper.set(context)
     try do
-      module.run(payload, metadata, context)
+      module.run(decode_payload(payload), metadata, context)
```

Comment:
Can we decompress the payload only once per job execution?
For example we could simply add a local variable in `handle_cast/2` to hold the decompressed payload so that decoding steps in `log_prefix/3` and `do_run/4` are combined.

Concerning `do_abandon/4`, it's OK to decode on-demand since:
- it should be called on a very rare occasions,
- when called, it can be far apart from `do_run/4` in terms of time (up to 30min); it seems better not to keep the decompressed payload in the heap memory of `AsyncJobRunner`.

Reply:
fixed in 2da715895c07c68e7e3abe20419d994ca535231f